aws-cache-helpers/src/middleware/ddb-cache.ts

```diff
@@ -0,0 +1,106 @@
+import {
+  CacheGet,
+  CacheSet,
+  getLogger,
+  LogFormat,
+  LogLevel,
+  SimpleCacheClient,
+} from '@gomomento/sdk';
+import {normalizeKeysFromRequestValue} from '../internal/utils/dynamodb';
+import {
+  InitializeHandler,
+  InitializeHandlerArguments,
+  InitializeHandlerOutput,
+  InitializeMiddleware,
+  MetadataBearer,
+  Pluggable,
+} from '@aws-sdk/types';
+
+const CommandCacheAllowList = ['GetItemCommand'];
+
+export interface CacheMiddleWareOptions {
+  tableName: string;
+  momentoAuthToken: string;
+  defaultCacheTtl: number;
+  cacheName: string;
+}
+
+export const NewCachingMiddleware = (
+  options: CacheMiddleWareOptions
+): Pluggable<any, any> => ({
+  applyToStack: clientStack => {
+    clientStack.add(cachingMiddleware(options), {tags: ['CACHE']});
+  },
+});
+export function cachingMiddleware(
+  options: CacheMiddleWareOptions
+): InitializeMiddleware<any, any> {
+  const momento = new SimpleCacheClient(
+    options.momentoAuthToken,
+    options.defaultCacheTtl,
+    {
+      loggerOptions: {
+        level: LogLevel.INFO,
+        format: LogFormat.JSON,
+      },
+    }
+  );
+  const logger = getLogger('momentoCachingMiddleware');
+
+  return <Output extends MetadataBearer>(
+      next: InitializeHandler<any, Output>
+    ): InitializeHandler<any, Output> =>
+    async (
+      args: InitializeHandlerArguments<any>
+    ): Promise<InitializeHandlerOutput<Output>> => {
+      // Check if we should cache this command
+      if (CommandCacheAllowList.includes(args.constructor.name)) {
+        // TODO right now this assumes GetItemCommand only we should make this more flexible in long run to
```

Comment:
style nit: not a blocker: this `if` is really long and if you start adding more supported operations it is going to get unwieldy.  I would refactor out a method to handle all of the logic for `GetItemCommand`.

Also I think it's worth actually doing the check here to ensure that it's a GetItemComand before you call your new function.  I know it's technically redundant while your allowlist has a size of 1, but it'll be a much safer way to go if we do end up adding more.

Lastly, I don't really think this allow-list pattern is going to scale to other operations because they will have such different inputs / requirements.  But as long as your public API that you are exposing to users doesn't leak that allow list variable or anything related to it then we are not coupled with this implementation so it's fine.

Your call on whether to address any of the above prior to merge.

Reply:
I had started on this and was getting messy with typing. I wanted to try do this as fast follow why added TODO. I agree to w/ not flexible right now. I want to make this part as more of high level router to broken out handlers was how I was envisioning down the line. 

Was just struggling w/ typing I don't think AWS was totally intending to hijack response was why was having hard time breaking out. I wanted try get v0 published and just do quick sanity test consuming package and come back if we get interest or want to add like "query" cache or some other feature. Maybe we can pair on typing a bit of middle-ware as well would be helpful to get 2nd pair eyes on it and chat through a bit.

Reply:
that's fine.